File: examples/android/MSDKSampleAppJava/lib/src/main/java/msdk/java/utils/StorageUtils.java
package msdk.java.utils;

import android.content.Context;
import android.system.ErrnoException;
import android.system.Os;

import msdk.java.logger.Logger;

import static msdk.java.utils.CommonUtils.getRootDir;

public class StorageUtils {
    public static void configureStoragePermissions(Context context) {
        Logger.getInstance().i("Configuring storage permissions");
        try {
            Os.setenv("EXTERNAL_STORAGE", getRootDir(context), true);
            // When we restore data, then we are not calling createOneTimeInfo
            // and hence ca-crt is not written within app directory
            // since the logic to write ca cert checks for file existence
            // we won't have to pay too much cost for calling this function inside init
        } catch (ErrnoException e) {
            Logger.getInstance().e("Failed to set environment variable storage", e);
        }
    }
}

File: examples/android/MSDKSampleAppKotlin/app/src/main/java/msdk/kotlin/sample/utils/StorageUtils.kt
package msdk.kotlin.sample.utils

import android.content.Context
import android.content.ContextWrapper
import android.system.ErrnoException
import android.system.Os
import msdk.kotlin.sample.logger.Logger.Companion.instance
import java.io.File
import java.io.FileNotFoundException
import java.io.FileWriter
import java.io.IOException

object StorageUtils {
    fun configureStoragePermissions(context: Context) {
        instance.i("Configuring storage permissions")
        try {
            Os.setenv("EXTERNAL_STORAGE", CommonUtils.getRootDir(context), true)
            // When we restore data, then we are not calling createOneTimeInfo
            // and hence ca-crt is not written within app directory
            // since the logic to write ca cert checks for file existence
            // we won't have to pay too much cost for calling this function inside init
        } catch (e: ErrnoException) {
            instance.e(
                "Failed to set environment variable storage",
                e
            )
        }
    }
}